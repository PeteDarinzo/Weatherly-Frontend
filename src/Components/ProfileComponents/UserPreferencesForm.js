import { Typography } from "@mui/material";
import React, { useState } from "react";
import { useSelector } from "react-redux";
import RangeSlider from "./RangeSlider";
import { makeStyles } from "@mui/styles";
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import Stack from '@mui/material/Stack';
import FormGroup from '@mui/material/FormGroup';
import Switch from '@mui/material/Switch';
import Radio from '@mui/material/Radio';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';
import { useDispatch } from "react-redux";
import { fetchForecastFromAPI } from "../../Actions/actions";

const useStyles = makeStyles({
  box: {
    marginTop: 40,
  },
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});

const UserPreferencesForm = ({ minTemp, maxTemp, units, thunderstorm, drizzle, rain, snow, overcast, updateUserPreferences }) => {

  const classes = useStyles();
  const dispatch = useDispatch();
  const userData = useSelector(store => store.user);

  let { lat, lon } = userData;

  const initialState = {
    minTemp,
    maxTemp,
    units,
    thunderstorm,
    drizzle,
    rain,
    snow,
    overcast
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      minTemp: null,
      maxTemp: null,
      units: value
    }));
  }

  function handleTempChange(tempArray) {
    setFormData(formData => ({
      ...formData,
      minTemp: tempArray[0],
      maxTemp: tempArray[1]
    }));
  }

  function handleToggle(e) {
    setFormData({
      ...formData,
      [e.target.name]: e.target.checked
    });
  }

  function handleSubmit() {
    // refetch forecast in case units changed
    dispatch(fetchForecastFromAPI(lat, lon, formData.units));
    updateUserPreferences(formData);
  }

  return (
    <Box className={classes.box}>
      <Typography variant="h4">Change Watch Preferences</Typography>
      <Box className={classes.box} sx={{border: "2px solid white", padding: "10px"}}>
        <Stack spacing={2}>
          <Typography variant="h5">Instructions:</Typography>
          <Typography>1. Select units of measurement.</Typography>
          <Typography>2. Select temperature range. You will not be recommended days where the temperature is between the two sliders.</Typography>
          <Typography>3. Select your preferred weather conditions.</Typography>
          <Typography>4. Click "SAVE PREFERENCES".</Typography>
        </Stack>
      </Box>
      <FormControl>
        <Box className={classes.box}>
          <FormLabel id="units" sx={{ fontSize: "1.5rem" }}>Units</FormLabel>
          <RadioGroup
            aria-labelledby="units"
            defaultValue={formData.units}
            name="units"
            onChange={handleChange}
          >
            <FormControlLabel
              value="imperial"
              control={<Radio />}
              label="Imperial"
            />
            <FormControlLabel
              value="metric"
              control={<Radio />}
              label="Metric"
            />
          </RadioGroup>
        </Box>

        <Box className={classes.box}>
          <FormLabel
            component="legend"
            sx={{ fontSize: "1.5rem" }}>Temperatures:</FormLabel>
          <RangeSlider
            handleTempChange={handleTempChange}
            units={formData.units}
            vals={[formData.minTemp, formData.maxTemp]}
          />
        </Box>

        <Box className={classes.box}>
          <FormControl
            component="fieldset"
            variant="standard"
          >
            <FormLabel
              component="legend"
              sx={{ fontSize: "1.5rem" }}>Weather Conditions:</FormLabel>
            <FormGroup>
              <FormControlLabel
                control={
                  <Switch
                    checked={formData.thunderstorm}
                    onChange={handleToggle}
                    name="thunderstorm" />
                }
                label="Thunderstorm"
              />
              <FormControlLabel
                control={
                  <Switch
                    checked={formData.drizzle}
                    onChange={handleToggle}
                    name="drizzle" />
                }
                label="Drizzle"
              />
              <FormControlLabel
                control={
                  <Switch
                    checked={formData.rain}
                    onChange={handleToggle}
                    name="rain" />
                }
                label="Rain"
              />
              <FormControlLabel
                control={
                  <Switch
                    checked={formData.snow}
                    onChange={handleToggle}
                    name="snow" />
                }
                label="Snow"
              />
              <FormControlLabel
                control={
                  <Switch
                    checked={formData.overcast}
                    onChange={handleToggle}
                    name="overcast" />
                }
                label="overcast"
              />
            </FormGroup>
          </FormControl>
        </Box>

        <Button
          onClick={handleSubmit}
          variant="contained"
          color="secondary"
          sx={{ marginTop: "40px" }}>Save Preferences</Button>

      </FormControl >
    </Box>
  );
}

export default UserPreferencesForm;