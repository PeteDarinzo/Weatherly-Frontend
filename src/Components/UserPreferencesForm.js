import { Container, TextField, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import RangeSlider from "./RangeSlider";
import { countryList, countryListAlpha2, codeListAlpha2 } from "../countryData";
import { makeStyles } from "@mui/styles";
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import FormGroup from '@mui/material/FormGroup';
import Switch from '@mui/material/Switch';
import Button from '@mui/material/Button';

const useStyles = makeStyles({
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});

const UserPreferencesForm = ({ minTemp, maxTemp, units, thunderstorm, drizzle, rain, snow, overcast, updateUserPreferences }) => {

  const classes = useStyles();

  const initialState = {
    minTemp,
    maxTemp,
    units,
    thunderstorm,
    drizzle,
    rain,
    snow,
    overcast
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  function handleTempChange(tempArray) {
    setFormData(formData => ({
      ...formData,
      minTemp: tempArray[0],
      maxTemp: tempArray[1]
    }));
  }

  function handleToggle(e) {
    setFormData({
      ...formData,
      [e.target.name]: e.target.checked
    });
  }

  function handleSubmit() {
    updateUserPreferences(formData);
  }

  return (
    <FormControl>
      <FormLabel id="units">Units</FormLabel>
      <RadioGroup
        aria-labelledby="units"
        defaultValue={formData.units}
        name="units"
        onChange={handleChange}
      >
        <FormControlLabel value="imperial" control={<Radio />} label="Imperial" />
        <FormControlLabel value="metric" control={<Radio />} label="Metric" />
      </RadioGroup>
      <RangeSlider
        handleTempChange={handleTempChange}
        units={formData.units}
        vals={[minTemp, maxTemp]}
      />
      <FormControl component="fieldset" variant="standard">
        <FormLabel component="legend">Toggle Watch Preferences</FormLabel>
        <FormGroup>
          <FormControlLabel
            control={
              <Switch checked={formData.thunderstorm} onChange={handleToggle} name="thunderstorm" />
            }
            label="Thunderstorm"
          />
          <FormControlLabel
            control={
              <Switch checked={formData.drizzle} onChange={handleToggle} name="drizzle" />
            }
            label="Drizzle"
          />
          <FormControlLabel
            control={
              <Switch checked={formData.rain} onChange={handleToggle} name="rain" />
            }
            label="Rain"
          />
          <FormControlLabel
            control={
              <Switch checked={formData.snow} onChange={handleToggle} name="snow" />
            }
            label="Snow"
          />
          <FormControlLabel
            control={
              <Switch checked={formData.overcast} onChange={handleToggle} name="overcast" />
            }
            label="overcast"
          />
        </FormGroup>
      </FormControl>
      <Button onClick={handleSubmit} variant="contained" color="secondary">Save Preferences</Button>
    </FormControl >
  )
}

export default UserPreferencesForm;