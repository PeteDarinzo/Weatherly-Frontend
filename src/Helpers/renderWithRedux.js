import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}