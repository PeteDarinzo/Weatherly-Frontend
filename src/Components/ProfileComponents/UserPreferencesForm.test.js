import UserPreferencesForm from "./UserPreferencesForm";
import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";

it("renders user preferences form without crashing", () => {
  renderWithRedux(
    <MemoryRouter>
      <UserPreferencesForm minTemp={40} maxTemp={60} units="F" thunderstorm={false} drizzle={false} rain={false} snow={true} overcast={true} updateUserPreferences={() => { }} />
    </MemoryRouter>
  );
});


it("matches the user preferences form snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <UserPreferencesForm minTemp={40} maxTemp={60} units="F" thunderstorm={false} drizzle={false} rain={false} snow={true} overcast={true} updateUserPreferences={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
