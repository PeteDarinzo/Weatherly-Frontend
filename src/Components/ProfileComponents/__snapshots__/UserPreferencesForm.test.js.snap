// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the user preferences form snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-box-3 MuiBox-root css-0"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Change Watch Preferences
    </h4>
    <div
      class="makeStyles-box-3 MuiBox-root css-ss5w87"
    >
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Instructions:
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          1. Select units of measurement.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          2. Select temperature range. You will not be recommended days where the temperature is between the two sliders.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          3. Select your preferred weather conditions.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          4. Click "SAVE PREFERENCES".
        </p>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="makeStyles-box-3 MuiBox-root css-0"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1v4x9p0-MuiFormLabel-root"
          id="units"
        >
          Units
        </label>
        <div
          aria-labelledby="units"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="units"
                type="radio"
                value="imperial"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Imperial
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="units"
                type="radio"
                value="metric"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Metric
            </span>
          </label>
        </div>
      </div>
      <div
        class="makeStyles-box-3 MuiBox-root css-0"
      >
        <legend
          class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1v4x9p0-MuiFormLabel-root"
        >
          Temperatures:
        </legend>
        <div
          class="MuiBox-root css-kudy17"
        >
          <span
            class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-marked MuiSlider-trackInverted css-1aznpnh-MuiSlider-root"
          >
            <span
              class="MuiSlider-rail css-1xst22u-MuiSlider-rail"
            />
            <span
              class="MuiSlider-track css-1h4p1o1-MuiSlider-track"
              style="left: 100%; width: 0%;"
            />
            <span
              class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
              data-index="0"
              style="left: 0%;"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
              data-index="0"
              style="left: 0%;"
            >
              0°C
            </span>
            <span
              class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
              data-index="1"
              style="left: 25%;"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
              data-index="1"
              style="left: 25%;"
            >
              10°C
            </span>
            <span
              class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
              data-index="2"
              style="left: 50%;"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
              data-index="2"
              style="left: 50%;"
            >
              20°C
            </span>
            <span
              class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
              data-index="3"
              style="left: 75%;"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
              data-index="3"
              style="left: 75%;"
            >
              30°C
            </span>
            <span
              class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
              data-index="4"
              style="left: 100%;"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
              data-index="4"
              style="left: 100%;"
            >
              40°C
            </span>
            <span
              class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
              data-index="0"
              style="left: 100%; pointer-events: none;"
            >
              <input
                aria-label="Minimum distance shift"
                aria-orientation="horizontal"
                aria-valuemax="40"
                aria-valuemin="0"
                aria-valuenow="40"
                aria-valuetext="40°C"
                data-index="0"
                max="40"
                min="0"
                step="1"
                style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                type="range"
                value="40"
              />
              <span
                aria-hidden="true"
                class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
              >
                <span
                  class="MuiSlider-valueLabelCircle"
                >
                  <span
                    class="MuiSlider-valueLabelLabel"
                  >
                    40
                  </span>
                </span>
              </span>
            </span>
            <span
              class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
              data-index="1"
              style="left: 100%; pointer-events: none;"
            >
              <input
                aria-label="Minimum distance shift"
                aria-orientation="horizontal"
                aria-valuemax="40"
                aria-valuemin="0"
                aria-valuenow="40"
                aria-valuetext="40°C"
                data-index="1"
                max="40"
                min="0"
                step="1"
                style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                type="range"
                value="40"
              />
              <span
                aria-hidden="true"
                class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
              >
                <span
                  class="MuiSlider-valueLabelCircle"
                >
                  <span
                    class="MuiSlider-valueLabelLabel"
                  >
                    40
                  </span>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        class="makeStyles-box-3 MuiBox-root css-0"
      >
        <fieldset
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <legend
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1v4x9p0-MuiFormLabel-root"
          >
            Weather Conditions:
          </legend>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    name="thunderstorm"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Thunderstorm
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    name="drizzle"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Drizzle
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    name="rain"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Rain
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    checked=""
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    name="snow"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Snow
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    checked=""
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    name="overcast"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                overcast
              </span>
            </label>
          </div>
        </fieldset>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-s5ox8d-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Save Preferences
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
